File: src/python/scraper/videos/base/video_scraper.py
from .base import BaseScraper
from .config import config

class VideoScraper(BaseScraper):
    def scrape(self, endpoint_name: str, params: dict = None) -> str:
        endpoint = config.get_endpoint(endpoint_name, params.get('page') if params else None)
        if not endpoint:
            raise ValueError(f"Endpoint '{endpoint_name}' not found in configuration.")
        soup = self.get_soup(endpoint, params)
        return soup.prettify()
    
    

File: src/api/videoScraper.ts
import BaseScraper from "./base";
import config from "./config/config";

abstract class VideoScraper extends BaseScraper {
  scrape(endpointName: string, params: Record<string, any>): Promise<string> {
    const page: number | null = params.page;
    const endpoint = config.getEndpoint(endpointName, page);
    if (!endpoint) throw new Error("Invalid endpoint " + endpointName);
    console.log(endpoint);
    const soup = this.getSoup(endpoint, params);
    return soup;
  }
}

export default VideoScraper;
